### PR TITLE
[SDK] `consumeProtectedData()`

### PR DESCRIPTION
**Pull request by Le-Caignec**

### CODE REVIEW

File: packages/sdk/src/config/config.ts [hunk 1]
```diff
@@ -16,9 +16,13 @@ export const DEFAULT_CONTRACT_ADDRESS =
   '0x3a4Ab33F3D605e75b6D00A32A0Fa55C3628F6A59'.toLowerCase();
 
 // iExec Dataset Registry
-export const POCO_REGISTRY_CONTRACT_ADDRESS =
+export const POCO_DATASET_REGISTRY_CONTRACT_ADDRESS =
   '0x799daa22654128d0c64d5b79eac9283008158730'.toLowerCase();
 
+// iExec Dataset Registry
```
Comment (cedric25) [inline packages/sdk/src/config/config.ts:22]: Comment to be updated I guess

File: packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+  if (protectedData.collection?.owner?.id !== userAddress) {
+    throw new ErrorWithData(
+      'This protected data is not part of a collection owned by the user.',
+      {
+        protectedDataAddress,
+        currentCollectionOwnerAddress: protectedData.collection?.owner?.id,
+      }
+    );
+  }
```
Comment (cedric25) [inline packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts:143]: Test irrelevant here?

It's fine if the protected data is in another collection, you could still consume it if you have rented it for example.

It's also fine if the protected data is in a collection of yours, you might want to consume your own protected data, anytime and for free.

File: packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+  // TODO: remove & set somewhere else
+  const hasActiveSubscriptions = protectedData.collection.subscriptions.some(
+    (subscription) => subscription.subscriber.id === userAddress
+  );
+  if (!hasActiveSubscriptions) {
+    throw new ErrorWithData('This collection has not valid subscription', {
+      collectionId: protectedData.collection.id,
+      currentCollectionOwnerAddress: protectedData.collection.owner?.id,
+    });
+  }
```
Comment (cedric25) [inline packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts:179]: Probably to remove, wrongly copy-pasted from some other method :)

File: packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    const workerpoolOrderbook = await iexec.orderbook.fetchWorkerpoolOrderbook({
+      workerpool: WORKERPOOL_ADDRESS,
+      app: appAddress,
+      dataset: vProtectedDataAddress,
+      minTag: SCONE_TAG,
+      maxTag: SCONE_TAG,
+    });
+    const workerpoolOrder = workerpoolOrderbook.orders[0]?.order;
+    if (workerpoolOrder.workerpoolprice > 0) {
+      throw new Error('No workerpool order free');
```
Comment (cedric25) [inline packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts:78]: Not sure to understand this one, what does it mean? Can we rephrase the error to make it more readable for a builder?

File: packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    }
+
```
Comment (cedric25) [inline packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts:80]: Before reaching out to the smart-contract, should we check in the subgraph:
 - The user has an active rental for this protected data

or

 - The user has an active subscription to the collection to which the protected data belongs

File: packages/sdk/src/lib/dataProtectorSharing/subgraph/getCollectionById.ts [hunk 6]
```diff
@@ -22,6 +24,12 @@ export async function getCollectionById({
           duration
           price
         }
+        subscriptions(orderBy: endDate, orderDirection: desc, where: { endDate_gte: "${today}" }){
+          endDate
+          subscriber {
+            id
+          }
+        }
```
Comment (cedric25) [inline packages/sdk/src/lib/dataProtectorSharing/subgraph/getCollectionById.ts:32]: See if that's really necessary once checks before consuming have been finalized.

**Review by cedric25 — CHANGES_REQUESTED** [review]
Some comments and questions.

➕ Start to write tests? Even if they're commented out at first.

File: packages/sdk/src/config/config.ts (hunk 1 above)
Comment (Le-Caignec) [reply packages/sdk/src/config/config.ts]: Commit: 238cf8a10453c058a91bc07c72b722a6f95c8151

File: packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts (hunk 2 above)
Comment (Le-Caignec) [reply packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts:143]: yeah you are right my mistake

File: packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts (hunk 3 above)
Comment (Le-Caignec) [reply packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts]: Commit: d8640a6f17f887529b43a04a56647e9efcd5cb32

File: packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+  const userAddress = (await iexec.wallet.getAddress()).toLowerCase();
+  const protectedData = await checkAndGetProtectedData({
+    graphQLClient,
+    protectedDataAddress: vProtectedDataAddress,
+    userAddress,
+  });
+
+  try {
+    const contentPath = '';
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply packages/sdk/src/lib/dataProtectorSharing/consumeProtectedData.ts]: Commit: 0a51332b5e9cd0b0797b7bf1dc1fa52f19f5c9a1
Fix ✅

**Review by cedric25 — APPROVED** [review]

**Pull request merged**